src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -460,8 +460,20 @@ private void verifyUserAccess(String jobId, AuthenticatedUserRequest userReq) {
       FlightMap inputParameters = flightState.getInputParameters();
       String flightSubjectId =
           inputParameters.get(JobMapKeys.SUBJECT_ID.getKeyName(), String.class);
+
```

Comment:
I forget where we landed around listing jobs.  I see this PR is focused on individual job retrieval.  I wonder if we could do some kind of filtering when retrieving the jobs (the job service would need to know how to build the filter clause for the different types of jobs but it might work?)

Reply:
I am glad you mentioned this!  I do think that ultimately we will want to be able to list jobs that the caller has access to (perhaps with some ability to filter as a future enhancement). Otherwise I don't think our vision of a "jobs status board" in the TDR UI will be fully realized.

My first instinct was that our users would still derive value from expanded access to jobs on demand even if that access is not reflected in job enumeration… but I think to later expand access for job enumeration would be a fair bit of refactoring and would be better to handle together.